cli/mender-artifact/partition.go

```diff
@@ -105,9 +110,23 @@ func newSDImgFile(fpath string, modcands []partition) (sdimgFile, error) {
 		rootfsa,
 		rootfsb,
 	}
+	delPartition = []int{0, 3}
 	return ps, nil
 }
 
+// cleanupUnusedPartitions removes the unused tmpfiles from /tmp,
+// but does not report any errors.
+// usedPartition: nil -- cleans up all the tmp-files given.
+func cleanupUnusedPartitions(partitions []partition, unusedPartition ...int) {
```

Comment:
I don't like how you are passing the `unusedPartition`. This indicates you are expecting multiple `int` parameters, which is OK, but in fact, you are passing a slice with the numbers of partitions to delete. What is more, why can't you store a list with the paths of the files to delete? We will eliminate looping and this info is accessible from where you are calling this function.